test/tests/bcs/ad_function_dirichlet_bc/tests

```diff
@@ -0,0 +1,19 @@
+[Tests]
+  issues = '#13261'
+  design = 'bcs/ADFunctionDirichletBC.md'
+  [./test]
+    type = 'Exodiff'
+    input = 'test.i'
+    exodiff = 'test_out.e'
+    requirement = "MOOSE shall support AD Dirichlet type boundary conditions defined by a function."
```

Comment:
I tried to pull the requirement up to the top block and just do 'detail' here @aeslaughter, but I take it you can't do that unless you are in a group? It's hard to think of a better group name without duplicating `test` three times?

Reply:
Correct. The group name should be descriptive of the group. Then the detail can just be the differences (or detail) that each test adds.  Also, the NQA-1 consultant recommended that we avoid saying "MOOSE shall" and prefer "The system shall" instead.